compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -531,6 +538,13 @@ pub fn compile_declarative_macro(
         None => {}
     }
 
+    // Compute the spans of the macro rules
+    let rule_spans = lhses
+        .iter()
+        .zip(rhses.iter())
+        .map(|(lhs, rhs)| lhs.span().to(rhs.span()))
+        .collect::<Vec<_>>();
```

Comment:
Can this just be the left hand sides? Ideally we don't want to highlight a large code block. Perhaps you should add a few tests that have a rhs that spans over many lines?

Reply:
Oh indeed, for the `unused_macros` lint a similar change was conducted in #90761. For the tests, it should already be tested due to the `^^^^`s in the snippets. E.g. it prints this:

```
error: 2nd rule of macro `num` is never used
  --> $DIR/unused-macro-rules-decl.rs:9:5
   |
LL |     (two) => { 2 },
   |     ^^^^^

```

To be on the safe side though, I've changed one test each to have an rhs with multiple lines.